### PR TITLE
docs: add message brokers section PEM-6141

### PR DESCRIPTION
**Pull request by addetz**
## Describe the Change

This PR adds a message brokers section explaining the HA and scaling of Palette message brokers.

## Changed Pages

💻 [Architecture](https://deploy-preview-4818--docs-spectrocloud.netlify.app/architecture/architecture-overview/#message-brokers)

## Jira Tickets

🎫 [PEM-6141](https://spectrocloud.atlassian.net/browse/PEM-6141)

## Backports

Can this PR be backported?

- [x] No. _Please leave a short comment below about why this PR cannot be backported._
Merging to release branch.

[PEM-6141]: https://spectrocloud.atlassian.net/browse/PEM-6141?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/docs-content/architecture/architecture-overview.md [hunk 1]
```diff
@@ -87,3 +87,29 @@ be optionally downloaded from a self-hosted private repository instead of pullin
 repository.
 
 ![Self-hosted Palette architecture diagram](/architecture_architecture-on-prem-detailed.webp)
+
+### Message Brokers
+
+Palette requires reliable, scalable and secure communication. The internal microservices use a Publish-Subscribe pattern
```
Comment (benradstone) [inline docs/docs-content/architecture/architecture-overview.md:93]:
```suggestion
Palette requires reliable, scalable, and secure communication. The internal microservices use a Publish-Subscribe pattern
```

File: docs/docs-content/architecture/architecture-overview.md [hunk 2]
```diff
@@ -87,3 +87,29 @@ be optionally downloaded from a self-hosted private repository instead of pullin
...
+implemented with [gRPC](https://grpc.io/) to achieve this. In order to support this communication pattern, a message
+broker service acts as the central hub for message exchange. Palette message brokers are automatically scaled, ensuring
+that a quorum is available for each cluster. The broker system is designed to provide the following functionality.
+
+1. It efficiently distributes incoming gRPC requests across multiple replicas of the message broker to optimize resource
+   usage and platform performance. This capability supports Palette's ability to manage large, enterprise Kubernetes
```
Comment (benradstone) [inline docs/docs-content/architecture/architecture-overview.md:99]:
```suggestion
   usage and platform performance. This capability supports Palette's ability to manage large enterprise Kubernetes
```

Comment (benradstone) [reply docs/docs-content/architecture/architecture-overview.md:99]: This is not grammatically incorrect, but I just think it reads better without the comma.

**Review by benradstone — APPROVED** [review]
Just two minor suggestions.

File: docs/docs-content/architecture/architecture-overview.md [hunk 3]
```diff
@@ -87,3 +87,29 @@ be optionally downloaded from a self-hosted private repository instead of pullin
...
+   clusters.
+2. It provides high availability by enabling clients to fail over to alternative replicas in the case of a pod failure.
+   By default, two replicas of the message broker are created in each cluster.
+3. It automatically adjusts to changes in the number of broker replicas without manual reconfiguration, ensuring that
+   the platform dynamically scales in response to load changes.
+4. It enforces message authentication and security by generating secondary certificates used for broker to broker
+   communication. This provides security in depth.
+
+Any Enterprise and VerteX Palette cluster will have a message broker that you can inspect. First, ensure that you can
+connect to the cluster. Refer to the
```
Comment (cloudmaniac) [inline docs/docs-content/architecture/architecture-overview.md:109]: That's only true for self-hosted. For multi-tenant and dedicated SaaS, they won't be able to do that. I know this paragraph is in the self-hosted section, but I still think we should make the distinction clear here.

File: docs/docs-content/architecture/architecture-overview.md [hunk 4]
```diff
@@ -87,3 +87,29 @@ be optionally downloaded from a self-hosted private repository instead of pullin
 repository.
 
 ![Self-hosted Palette architecture diagram](/architecture_architecture-on-prem-detailed.webp)
+
+### Message Brokers
+
+Palette requires reliable, scalable and secure communication. The internal microservices use a Publish-Subscribe pattern
+implemented with [gRPC](https://grpc.io/) to achieve this. In order to support this communication pattern, a message
+broker service acts as the central hub for message exchange. Palette message brokers are automatically scaled, ensuring
+that a quorum is available for each cluster. The broker system is designed to provide the following functionality.
```
Comment (cloudmaniac) [inline docs/docs-content/architecture/architecture-overview.md:96]: It's unclear which "cluster" we are referring to here. AFAIK, message brokers are only running in the Palette management plane.

File: docs/docs-content/architecture/architecture-overview.md [hunk 5]
```diff
@@ -87,3 +87,29 @@ be optionally downloaded from a self-hosted private repository instead of pullin
...
+
+1. It efficiently distributes incoming gRPC requests across multiple replicas of the message broker to optimize resource
+   usage and platform performance. This capability supports Palette's ability to manage large, enterprise Kubernetes
+   clusters.
```
Comment (cloudmaniac) [inline docs/docs-content/architecture/architecture-overview.md:100]: How to say also a "larger number of K8s clusters"?

File: docs/docs-content/architecture/architecture-overview.md [hunk 6]
```diff
@@ -87,3 +87,29 @@ be optionally downloaded from a self-hosted private repository instead of pullin
...
+2. It provides high availability by enabling clients to fail over to alternative replicas in the case of a pod failure.
+   By default, two replicas of the message broker are created in each cluster.
```
Comment (cloudmaniac) [inline docs/docs-content/architecture/architecture-overview.md:102]: Same comment as above: it's unclear which cluster we are referring to here

File: docs/docs-content/architecture/architecture-overview.md (hunk 3 above)
Comment (addetz) [reply docs/docs-content/architecture/architecture-overview.md]: Clarified now. Thanks so much!

File: docs/docs-content/architecture/architecture-overview.md (hunk 4 above)
Comment (addetz) [reply docs/docs-content/architecture/architecture-overview.md]: Clarified all cluster instances now. Thanks so much!

**Pull request merged**